### DISCUSSION TITLE
Psycopg dependency isn't Python3.9 compatible

### DISCUSSION THREAD

**Discussion by EBoisseauSierra** [apache/superset#14181] [General]
Even if I understand Python 3.9 isn't officially supported, I tried to install `1.1.0` on it to run the tests. I failed to install the dependencies due to outdated `psycopg` version.

Indeed, we require [`psycopg2-binary==2.8.5`](https://github.com/apache/superset/blob/master/requirements/development.txt#L26) while from [this comment](https://github.com/psycopg/psycopg2/issues/1200#issuecomment-735237051):

> Binary packages for Python 3.9 for linux and windows are available only from 2.8.6 on.

### Expected results

Be able to install the testing dependencies on Python 3.9.

```
git checkout 1.1.0
python3.9 -m venv venv && source venv/bin/activate
pip install -r requirements/testing.txt
```

### Actual results

```
$ pip install -r requirements/testing.txt

[…]

Collecting psycopg2-binary==2.8.5
  Downloading https://files.pythonhosted.org/packages/97/00/ed4c82364741031d745867f83820d4f373aa891098a5785841850491c9ba/psycopg2-binary-2.8.5.tar.gz (381kB)
     |████████████████████████████████| 389kB 61.0MB/s 
    ERROR: Command errored out with exit status 1:
     command: /home/fedora/superset/venv/bin/python3.9 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-8hhvthd7/psycopg2-binary/setup.py'"'"'; __file__='"'"'/tmp/pip-install-8hhvthd7/psycopg2-binary/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info
         cwd: /tmp/pip-install-8hhvthd7/psycopg2-binary/
    Complete output (23 lines):
    running egg_info
    creating /tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info/psycopg2_binary.egg-info
    writing /tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info/psycopg2_binary.egg-info/PKG-INFO
    writing dependency_links to /tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info/psycopg2_binary.egg-info/dependency_links.txt
    writing top-level names to /tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info/psycopg2_binary.egg-info/top_level.txt
    writing manifest file '/tmp/pip-install-8hhvthd7/psycopg2-binary/pip-egg-info/psycopg2_binary.egg-info/SOURCES.txt'
    
    Error: pg_config executable not found.
    
    pg_config is required to build psycopg2 from source.  Please add the directory
    containing pg_config to the $PATH or specify the full executable path with the
    option:
    
        python setup.py build_ext --pg-config /path/to/pg_config build ...
    
    or with the pg_config option in 'setup.cfg'.
    
    If you prefer to avoid building psycopg2 from source, please install the PyPI
    'psycopg2-binary' package instead.
    
    For further information please check the 'doc/src/install.rst' file (also at
    <https://www.psycopg.org/docs/install.html>).
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

### Environment

(please complete the following information):

- superset version: `1.1.0`
- python version: `3.9.4`

### Checklist

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Suggested solution

Bump the `psycopg2-binary` requirement to ≥ `2.8.6`. (Note: this isn't the only Py39 incompatible dependency!)

**Comment by EBoisseauSierra** [comment]
OT: Next in line is [`pillow==7.2.0`](https://github.com/apache/superset/blob/master/requirements/development.txt#L25) — as Py3.9 is [supported from 8.*](https://stackoverflow.com/a/64254835/5433628) onwards.

**Comment by junlincc** [comment]
@EBoisseauSierra moving it to Discussion. feel free to open a PR:)

**Comment by rusackas** [comment]
@EBoisseauSierra is this still an issue, or should we remove this thread?
